api/src/main/java/uk/gov/dhsc/htbhf/card/service/CardService.java

```diff
@@ -43,12 +40,28 @@ public CardResponse createCard(CardRequestDTO cardRequest) {
      * @return The response
      */
     public DepositFundsResponse depositFunds(String cardId, DepositFundsRequestDTO depositFundsRequest) {
-        String cardUrl = buildCardUrl(cardId);
-        ResponseEntity<DepositFundsResponse> response = restTemplate.postForEntity(cardUrl, depositFundsRequest, DepositFundsResponse.class);
+        String depositUrl = buildDepositUrl(cardId);
+        ResponseEntity<DepositFundsResponse> response = restTemplate.postForEntity(depositUrl, depositFundsRequest, DepositFundsResponse.class);
         return response.getBody();
     }
 
-    private String buildCardUrl(String cardId) {
+    /**
+     * Call through to the card service provider to get the card's balance.
+     *
+     * @param cardId The card id
+     * @return The balance response
+     */
+    public CardBalance getBalance(String cardId) {
+        String balanceUrl = buildBalanceUrl(cardId);
+        ResponseEntity<CardBalance> response = restTemplate.getForEntity(balanceUrl, CardBalance.class);
+        return response.getBody();
+    }
+
+    private String buildBalanceUrl(String cardId) {
+        return uri + "/" + cardId + "/balance";
```

Comment:
Can we refactor the first part `uri + "/" + cardId ` into a method to avoid duplication?

Reply:
Done.